generators/typescript/sdk/CHANGELOG.md

```diff
@@ -5,6 +5,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.40.2] - 2024-09-12
+
+- Fix: The generated SDK now handles reading IR JSONs that are larger than 500MB. In order to 
+  to this, the function `streamObjectFromFile` is used instead of `JSON.parse`. 
```

Comment:
I only released the TypeScript SDK generator since that is the one a user complained about

Reply:
"Complained" is a bit of a harsh choice of words perhaps – I reported that your software doesn't work properly with large specs, and then offered up assistance in solving the bugs.

Nevertheless, I can confirm this does work with our spec now. I've also checked the size of the `ir.json` and it's 170 356 624 bytes (170,4 MB on disk) – so significantly smaller than 500 MB. I believe the limit isn't file size, but allocation when parsing using `JSON.parse`, which is obviously not the same as the file size.

Presumably there's some correlation between file size and memory allocation when parsing, but I doubt it's a strict 1:1 mapping. I wouldn't be surprised if parsing could exceed the limit even with smaller files than our case. Likely not a lot smaller, but still. But I digress.

Streaming the parsing and the writing should hopefully be a solid fix, thank you for looking into this. We also use the Python and Go generators, so if those could be updated as well that would be much appreciated – thank you! 🙏

Reply:
Apologies, didn't mean to offend you! And tbh totally reasonable to be frustrated by Fern having this bug.